### PR TITLE
Do not build netstandard2.0

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
## Improve CI performance.

We do not use `netstandard2.0` builds and we do not ship them. So we don't need to wait for them to be built and analyzed.

With this change, we also unit test the build that we ship.

## Summary 

`Build and package` is 41% faster. Making every build 0:47 shorter in average.

`.NET Analysis / Run .Net code analysis - before` is 28% faster (0:40), but it doesn't help with total CI run too much yet, because Java ITs take longer anyway. And we already know how to speed them up.

## Details

**Build and package - before**
Min: 1:07
Max: 3:30
Average: 1:52

_Side note:_ Why the hell do we have such volatile differences on the same VMs anyway?

**Build and package - before**
Min: 0:52
Max: 1:47
Average: 1:05

**.NET Analysis / Run .Net code analysis - before**
Min: 2:11
Max: 2:58
Average: 2:25

**.NET Analysis / Run .Net code analysis - after**
Min: 1:35
Max: 1:59
Average: 1:44

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Extensions/ITupleOperationWrapperExtensionsTests.cs [hunk 1]
```diff
@@ -60,8 +60,8 @@ static string DeeplyNestedTuple(int depth)
         public void AllElements_Performance_LargeTuple()
         {
             var largeTuple = LargeTuple(500); // (1, 2,... , 500)
-            // Actual execution time is about 0.5ms.
-            AssertAllElementsExecutionTimeBeLessThan(largeTuple, 5.Milliseconds());
+            // Actual execution time is about 0.4ms - 0.7ms
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Extensions/ITupleOperationWrapperExtensionsTests.cs:63]: 2x out of 10 runs failed with `0.007s and 327.3µs`

File: analyzers/tests/SonarAnalyzer.UnitTest/Extensions/ITupleOperationWrapperExtensionsTests.cs [hunk 2]
```diff
@@ -60,8 +60,8 @@ static string DeeplyNestedTuple(int depth)
...
+            AssertAllElementsExecutionTimeBeLessThan(largeTuple, 10.Milliseconds());
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Extensions/ITupleOperationWrapperExtensionsTests.cs:64]: 2x out of 10 runs failed with `0.007s and 327.3µs`

File: azure-pipelines.yml [hunk 3]
```diff
@@ -87,7 +87,7 @@ stages:
           solution: '$(solution)'
           platform: '$(buildPlatform)'
           configuration: '$(buildConfiguration)'
-          msbuildArgs: '/p:DeployExtension=false /p:SignAssembly=$(isReleaseBranch) /p:AssemblyOriginatorKeyFile="$(snk.secureFilePath)" /p:Sha1=$(Build.SourceVersion) /p:BuildNumber=$(Build.BuildId) /p:WarningLevel=0'
+          msbuildArgs: '/p:DeployExtension=false /p:SignAssembly=$(isReleaseBranch) /p:AssemblyOriginatorKeyFile="$(snk.secureFilePath)" /p:Sha1=$(Build.SourceVersion) /p:BuildNumber=$(Build.BuildId) /p:WarningLevel=0 /p:BuildOnlyNet46=true'
```
Comment (costin-zaharia-sonarsource) [inline azure-pipelines.yml:90]: This works on the pipeline since we called `nuget restore` with the right params already. Right?

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Comment by pavel-mikula-sonarsource** [comment]
Status Update: Issue https://github.com/NuGet/Home/issues/12010 was confirmed by NuGet

File: azure-pipelines.yml (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply azure-pipelines.yml:90]: Yes, exactly.

**Pull request merged**